docs/source/parsing_best_practices.md

```diff
@@ -0,0 +1,240 @@
+<cudf.Series nrows=4050000 >
+```
+
+*We see we can get a `14x` boost if we sort our strings by length*
+ 
+
+## Character and Substring Substitution/Removal
+
+
+A lot of times during parsing you want to remove substrings and characters . For example, In our post, [show me the word count](https://medium.com/rapids-ai/show-me-the-word-count-3146e1173801), we needed to remove a long list of punctuation characters, as we didn’t want the punctuation characters to impact unique string (word) counts. 
+
+
+
+Let’s look at various options of removing punctuation characters on the below example:
+
+```python
+import random
+import cudf
+import re
+
+small_string = "word1!word2,word3<word4>-word5>word7~word8+word9:word10[word11]"
+long_string =  small_string*100
+example_list = [small_string]*5_000_000 + [long_string]*50_000
+random.shuffle(example_list)
+
+cudf_sr = cudf.Series(example_list)
+
+filters = ['!','_',',','{','}','<','>','~','+',':','[',']','-']
+```
+
+### Using a for loop:
+
+We could replace them one-by-one in many calls to `replace`, a common first approach to parsing semi-structured logs, but that would be a lot of repetitive function calls and can be slow. 
+
+```python
+%%time
+for filter_char in list(map(re.escape,filters)):
+    cudf_sr = cudf_sr.str.replace(filter_char, ' ')
+```
+
+```python
+CPU times: user 8.25 s, sys: 2.39 s, total: 10.6 s
+Wall time: 10.7 s
+```
+
+
+### Using regex:
+We can also use a regular expression (regex) for the replacement. 
+
+
+#### Regex with capturing groups:
+
+Our first thought may be to use a pattern like  `(!)|(,)...|(%)`    for the character replacement.
+
+```python
+filters_regex_capturing_group = '|'.join([f'({ch})' for ch in  map(re.escape, filters)])
+print(filters_regex_capturing_group)
+%time cudf_sr = cudf_sr.str.replace(filters_regex_capturing_group, ' ')
+```
+
+```python
+(!)|(_)|(,)|(\{)|(\})|(<)|(>)|(\~)|(\+)|(:)|(\[)|(\])|(\-)
+CPU times: user 17.8 s, sys: 5.51 s, total: 23.3 s
+Wall time: 23.3 s
+```
+
+#### Regex without capturing groups:
+
+Above we used [capturing groups](https://www.regular-expressions.info/brackets.html) to do the regex replacment which we  dont really need when doing a replacment without back references. If we use non capturing groups instead we can get almost a 1.75x  boost in our example: 
+
+
+```python
+filters_regex_without_capturing_group =  '|'.join([f'(?:{ch})' for ch in  map(re.escape, filters)])
+print(filters_regex_without_capturing_group)
+%time cudf_sr = cudf_sr.str.replace(filters_regex_without_capturing_group, ' ')
+```
+
+```python
+(?:!)|(?:_)|(?:,)|(?:\{)|(?:\})|(?:<)|(?:>)|(?:\~)|(?:\+)|(?::)|(?:\[)|(?:\])|(?:\-)
+CPU times: user 10.1 s, sys: 3.19 s, total: 13.3 s
+Wall time: 13.3 s
+```
+
+
+
+### Using `nvtext`
+
+`nvtext` provides out-of-the-box functions which make these computations like removing multiple characters and stop words much faster.
+
+
+```python
+%time cudf_sr = cudf_sr.str.replace_multi(filters, ' ',regex=False)
```

Comment:
The `replace_multi` is an `nvstrings` method and not an `nvtext` method. The `nvtext` method is `replace_tokens` and uses delimiters to locate tokens. The `nvstrings` `replace_multi` replaces strings and not tokens.

Reply:
Yup, this was incorrect, this should just have been `nvstrings`.replace_multi .

For `nvtext.replace_tokens`, I  can create another example , where i remove `stop-words` and show different regex  performance for that . 

https://rapidsai.github.io/projects/nvstrings/en/0.9.0/api.html?source=post_page---------------------------#nvtext.replace_tokens

Reply:
Changed this to add different  examples for both,  `nvstrings.replace_multi` and `nvtext.replace_tokens`